lib/galaxy/datatypes/data.py

```diff
@@ -1205,6 +1205,143 @@ def regex_line_dataprovider(
 class Directory(Data):
     """Class representing a directory of files."""
 
+    # The behavior of this class is intended to be similar to that of
```

Comment:
It should really just be a directory, can you make all that fancy stuff a subclass and include some specific implementation ? We're making heavy use of this in the CWL branch. ee also https://github.com/galaxyproject/galaxy/compare/dev...mvdbeek:galaxy:directory_datatype_improvements for details.

Reply:
well, what exactly do you mean? It really *is* just a directory, but one with minimal knowledge of displaying its contents and size? Why would that be a problem for the CWL branch?

But sure, I can turn it into a SmartDirectory subclass or something if you prefer, I would just like to understand why that's an advantage.

Reply:
>  but that won't work with subclasse

There's surely a better way than repeating the code.

> Why would that be a problem for the CWL branch?

It's not a problem for the CWL branch, but I don't think we should store that in the database. Literal filesystem directories don't have this, and the total file size is redundant with what we calculate and store anyway.

Reply:
And they can, as for any other file.

Reply:
how?

Reply:
It works. Thanks for the hint!

Reply:
added the file_ext